service/resource/legacy/resource.go

```diff
@@ -0,0 +1,372 @@
+package legacy
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/cenk/backoff"
+	"github.com/giantswarm/flanneltpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/flannel-operator/service/etcdv2"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "legacy"
+)
+
+// Config represents the configuration used to create a new config map resource.
+type Config struct {
+	BackOff   backoff.BackOff
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+	Store     *etcdv2.Service
+
+	EtcdCAFile  string
+	EtcdCrtFile string
+	EtcdKeyFile string
+}
+
+// DefaultConfig provides a default configuration to create a new config map
+// resource by best effort.
+func DefaultConfig() Config {
+	return Config{
+		BackOff:   nil,
+		K8sClient: nil,
+		Logger:    nil,
+		Store:     nil,
+
+		EtcdCAFile:  "",
+		EtcdCrtFile: "",
+		EtcdKeyFile: "",
+	}
+}
+
+// Resource implements the config map resource.
+type Resource struct {
+	backOff   backoff.BackOff
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+	store     *etcdv2.Service
+
+	etcdCAFile  string
+	etcdCrtFile string
+	etcdKeyFile string
+}
+
+// New creates a new configured config map resource.
+func New(config Config) (*Resource, error) {
```

Comment:
This here is the legacy resource. We can now use the reconciler framework while still executing the old code. This helps for the migration to proper resource implementations.